ui/common/src/socket.ts

```diff
@@ -90,49 +93,44 @@ export default class StrongSocket {
         ...(settings.params || {}),
       },
     };
-    const customPingDelay = storedIntProp('socket.ping.interval', 2500)();
-
-    this.options = {
-      ...StrongSocket.defaultOptions,
-      ...(settings.options || {}),
-      pingDelay: customPingDelay > 400 ? customPingDelay : 2500,
-    };
     this.version = version;
     site.pubsub.on('socket.send', this.send);
     this.connect();
+    this.flushStats();
```

Comment:
`flushStats` is related to the socket test.

so is the `'pagehide'` handler. these are discussed below.